Explorer/Assets/Scripts/ECS/SceneLifeCycle/Systems/CheckCameraQualifiedForRepartitioningSystem.cs

```diff
@@ -25,24 +26,41 @@ public partial class CheckCameraQualifiedForRepartitioningSystem : BaseUnityLoop
     {
         private readonly IRealmData realmData;
         private readonly IPartitionSettings partitionSettings;
+        private readonly CameraSamplingData cameraSamplingData;
 
-        internal CheckCameraQualifiedForRepartitioningSystem(World world, IPartitionSettings partitionSettings, IRealmData realmData) : base(world)
+        internal CheckCameraQualifiedForRepartitioningSystem(World world, IPartitionSettings partitionSettings, IRealmData realmData, CameraSamplingData cameraSamplingData) : base(world)
         {
             this.partitionSettings = partitionSettings;
             this.realmData = realmData;
+            this.cameraSamplingData = cameraSamplingData;
         }
 
         protected override void Update(float t)
         {
             // it should be updated only if realm is already loaded
-            if (realmData.Configured)
-                CheckCameraTransformChangedQuery(World);
+            if (!realmData.Configured)
+                return;
+
+            cameraSamplingData.IsDirty = false;
+            ModifyCameraSamplingBasedOnTeleportIntentQuery(World);
+            CheckCameraTransformChangedQuery(World);
         }
 
         [Query]
         private void CheckCameraTransformChanged(ref CameraSamplingData cameraSamplingData, ref CameraComponent cameraComponent)
         {
+            if (cameraSamplingData.IsDirty)
+                return;
+
             ScenesPartitioningUtils.CheckCameraTransformChanged(cameraSamplingData, in cameraComponent, partitionSettings.PositionSqrTolerance, partitionSettings.AngleTolerance);
```

Comment:
PartitionDiscreteDataBase will be mutated in CheckCameraTransformChanged, please clarify the name with CQRS

Reply:
Done: https://github.com/decentraland/unity-explorer/pull/717/commits/7fd26b63d58f1d34237be46c1588f83b33b94794